src/test/java/seedu/task/model/task/DateTest.java

```diff
@@ -0,0 +1,57 @@
+package seedu.task.model.task;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.task.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class DateTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Date(null));
+    }
+
+    @Test
+    public void constructor_invalidDate_throwsIllegalArgumentException() {
+        String invalidDate = "invalid date";
+        assertThrows(IllegalArgumentException.class, () -> new Date(invalidDate));
+    }
+
+    @Test
+    public void isValidDate() {
```

Comment:
It might be better to split this into 2 different test methods
isValidDate_validInput_returnsTrue
and
isValidDate_invalidInput_returnsTrue

following test naming conventions.

Reply:
Great suggestion. My latest commit would have integrated your suggestion with it.